sdk/storage/azure-storage-blob-batch/src/main/java/com/azure/storage/blob/batch/BlobBatch.java

```diff
@@ -0,0 +1,427 @@
+            .buildAsyncClient();
+
+        this.batchOperationQueue = new ConcurrentLinkedDeque<>();
+        this.batchRequest = new ArrayList<>();
+        this.batchMapping = new ConcurrentHashMap<>();
+    }
+
+    /**
+     * Adds a delete blob operation to the batch.
+     *
+     * <p><strong>Code sample</strong></p>
+     *
+     * {@codesnippet com.azure.storage.blob.batch.BlobBatch.deleteBlob#String-String}
+     *
+     * @param containerName The container of the blob.
+     * @param blobName The name of the blob.
+     * @return a {@link Response} that will be used to associate this operation to the response when the batch is
+     * submitted.
+     * @throws UnsupportedOperationException If this batch has already added an operation of another type.
+     */
+    public Response<Void> deleteBlob(String containerName, String blobName) {
+        return deleteBlobHelper(String.format("%s/%s", containerName, blobName), null, null);
+    }
+
+    /**
+     * Adds a delete blob operation to the batch.
+     *
+     * <p><strong>Code sample</strong></p>
+     *
+     * {@codesnippet com.azure.storage.blob.batch.BlobBatch.deleteBlob#String-String-DeleteSnapshotsOptionType-BlobAccessConditions}
+     *
+     * @param containerName The container of the blob.
+     * @param blobName The name of the blob.
+     * @param deleteOptions Delete options for the blob and its snapshots.
+     * @param blobAccessConditions Additional access conditions that must be met to allow this operation.
+     * @return a {@link Response} that will be used to associate this operation to the response when the batch is
+     * submitted.
+     * @throws UnsupportedOperationException If this batch has already added an operation of another type.
+     */
+    public Response<Void> deleteBlob(String containerName, String blobName,
+        DeleteSnapshotsOptionType deleteOptions, BlobAccessConditions blobAccessConditions) {
+        return deleteBlobHelper(String.format("%s/%s", containerName, blobName), deleteOptions, blobAccessConditions);
+    }
+
+    /**
+     * Adds a delete blob operation to the batch.
+     *
+     * <p><strong>Code sample</strong></p>
+     *
+     * {@codesnippet com.azure.storage.blob.batch.BlobBatch.deleteBlob#String}
+     *
+     * @param blobUrl URI of the blob.
+     * @return a {@link Response} that will be used to associate this operation to the response when the batch is
+     * submitted.
+     * @throws UnsupportedOperationException If this batch has already added an operation of another type.
+     */
+    public Response<Void> deleteBlob(String blobUrl) {
+        return deleteBlobHelper(getUrlPath(blobUrl), null, null);
+    }
+
+    /**
+     * Adds a delete blob operation to the batch.
+     *
+     * <p><strong>Code sample</strong></p>
+     *
+     * {@codesnippet com.azure.storage.blob.batch.BlobBatch.deleteBlob#String-DeleteSnapshotsOptionType-BlobAccessConditions}
+     *
+     * @param blobUrl URI of the blob.
```

Comment:
These are actually URLs, that you support, the documentation is misleading.

https://danielmiessler.com/study/url-uri/

Reply:
Will update in Preview 5

Reply:
https://github.com/Azure/azure-sdk-for-java/issues/5756